source/common/config/grpc_mux_impl.cc

```diff
@@ -275,6 +289,38 @@ void GrpcMuxImpl::queueDiscoveryRequest(const std::string& queue_item) {
   drainRequests();
 }
 
+GrpcMuxImpl::ApiState& GrpcMuxImpl::apiStateFor(const std::string& type_url) {
+  auto itr = api_state_.find(type_url);
+  if (itr == api_state_.end()) {
+    auto ttl_expiry_callback = [this, type_url](const std::vector<std::string>& expired) {
+      // The TtlManager triggers a callback with a list of all the expired elements, which we need
+      // to compare against the various watched resources to return the subset that each watch is
+      // subscribed to.
+
+      // We convert the incoming list into a set in order to more efficiently perform this
+      // comparison when there are a lot of watches.
+      absl::flat_hash_set<std::string> all_expired;
+      all_expired.insert(expired.begin(), expired.end());
+
+      for (auto watch : apiStateFor(type_url).watches_) {
```

Comment:
Can you add a comment on the recursion here? Is there a way to just use concrete `api_state_` here since we know that it should be established in the map as a result of the enclosing method?